Publish fails with git error "not our ref"

**Describe the bug**
When I try to run the action `JamesIves/github-pages-deploy-action@releases/v3` it fails with the cryptic sounding error message:

> fatal: remote error: upload-pack: not our ref 4493cf249a1648be2ad365fe94ece337bfbcb5d9

I suspect this is related to my usage of submodules. Initially I suspected it was because I was using `git@` for the submodules, so I wrote a step just before the upload action that rewrites the URLs in `.gitmodules` using `sed`. This didn't seem to fix the issue. I can't figure out what the refs are pointing to. They are neither in the submodules, nor my main repository.

**Reproduce**
Run the workflow from https://github.com/fredefox/cat/pull/16 There you can also find the workflow that I'm using: https://github.com/fredefox/cat/blob/fredefox/gh-pages/.github/workflows/main.yml

**Logs**
```
2020-05-16T11:34:46.0657910Z ##[group]Run JamesIves/github-pages-deploy-action@releases/v3
2020-05-16T11:34:46.0658364Z with:
2020-05-16T11:34:46.0658684Z   ACCESS_TOKEN: ***
2020-05-16T11:34:46.0658763Z   BRANCH: gh-pages
2020-05-16T11:34:46.0659014Z   FOLDER: html
2020-05-16T11:34:46.0659280Z ##[endgroup]
2020-05-16T11:34:46.1899409Z Checking configuration and starting deployment… 🚦
2020-05-16T11:34:46.1912285Z Deploying using Access Token… 🔑
2020-05-16T11:34:46.1914504Z Configuring git…
2020-05-16T11:34:46.1991775Z [command]/usr/bin/git init
2020-05-16T11:34:46.2084724Z Reinitialized existing Git repository in /home/runner/work/cat/cat/.git/
2020-05-16T11:34:46.2141438Z [command]/usr/bin/git config user.name fredefox
2020-05-16T11:34:46.2200441Z [command]/usr/bin/git config user.email fredefox@users.noreply.github.com
2020-05-16T11:34:46.2258537Z [command]/usr/bin/git remote rm origin
2020-05-16T11:34:46.2328556Z [command]/usr/bin/git remote add origin https://***@github.com/fredefox/cat.git
2020-05-16T11:34:46.2387925Z [command]/usr/bin/git fetch
2020-05-16T11:34:46.7365950Z From https://github.com/fredefox/cat
2020-05-16T11:34:46.7368544Z  * [new branch]      abstract-funext   -> origin/abstract-funext
2020-05-16T11:34:46.7368818Z  * [new branch]      fredefox/gh-pages -> origin/fredefox/gh-pages
2020-05-16T11:34:46.7369072Z  * [new branch]      fredefox/travis   -> origin/fredefox/travis
2020-05-16T11:34:46.7370460Z  * [new branch]      gh-pages          -> origin/gh-pages
2020-05-16T11:34:46.7370878Z  * [new branch]      master            -> origin/master
2020-05-16T11:34:46.7371101Z  * [new branch]      try-wrapped-paths -> origin/try-wrapped-paths
2020-05-16T11:34:46.7540964Z  * [new tag]         0.0.1             -> 0.0.1
2020-05-16T11:34:46.7541648Z  * [new tag]         1.0.0             -> 1.0.0
2020-05-16T11:34:46.7542171Z  * [new tag]         1.0.1             -> 1.0.1
2020-05-16T11:34:46.7542657Z  * [new tag]         1.1.0             -> 1.1.0
2020-05-16T11:34:46.7543138Z  * [new tag]         1.2.0             -> 1.2.0
2020-05-16T11:34:46.7543626Z  * [new tag]         1.3.0             -> 1.3.0
2020-05-16T11:34:46.7544107Z  * [new tag]         1.4.0             -> 1.4.0
2020-05-16T11:34:46.7544564Z  * [new tag]         1.4.1             -> 1.4.1
2020-05-16T11:34:46.7545050Z  * [new tag]         1.5.0             -> 1.5.0
2020-05-16T11:34:46.7546510Z  * [new tag]         1.6.0             -> 1.6.0
2020-05-16T11:34:46.8730768Z Fetching submodule libs/agda-stdlib
2020-05-16T11:34:47.0713083Z Fetching submodule libs/cubical
2020-05-16T11:34:47.4210696Z fatal: remote error: upload-pack: not our ref 0d3f02e68297e940227137beac45fc1bce6e2bea
2020-05-16T11:34:47.7142770Z fatal: remote error: upload-pack: not our ref 4493cf249a1648be2ad365fe94ece337bfbcb5d9
2020-05-16T11:34:47.7215624Z Errors during submodule fetch:
2020-05-16T11:34:47.7215893Z 	libs/cubical
2020-05-16T11:34:47.7216327Z 	libs/agda-stdlib
2020-05-16T11:34:47.7257861Z ##[error]There was an error initializing the repository: The process '/usr/bin/git' failed with exit code 1 ❌
2020-05-16T11:34:47.7267482Z failed
2020-05-16T11:34:47.7269681Z Deployment Failed ❌
2020-05-16T11:34:47.7272095Z ::set-env name=DEPLOYMENT_STATUS::failed
```

I've made my change and commited `node_modules`, yet it seems to having some issues resolving the dependencies. Perhaps I'm doing something wrong? The error that I'm seeing:
```
##[error]Could not find file '/home/runner/work/_actions/_temp_7c50805b-74c8-4906-8020-8993a8e0bb09/_staging/fredefox-github-pages-deploy-action-639e288/node_modules/istanbul-lib-instrument/node_modules/@babel/core/node_modules/.bin/json5'.
```
From https://github.com/fredefox/cat/pull/16/checks?check_run_id=682675060

On this branch: https://github.com/fredefox/github-pages-deploy-action/tree/dev

--- a/src/git.ts
+++ b/src/git.ts
@@ -24,7 +24,7 @@ export async function init(action: ActionInterface): Promise<void | Error> {
       `git remote add origin ${action.repositoryPath}`,
       action.workspace
     )
-    await execute(`git fetch`, action.workspace)
+    await execute(`git fetch --no-recurse-submodules`, action.workspace)
 
     info('Git configured… 🔧')
   } catch (error) {

